### PR TITLE
MM-16925: fixed resp.Body leaks

### PR DESCRIPTION
**Pull request by levb**
- Cherry-picked https://github.com/mattermost/mattermost-plugin-jira/pull/214
- Fixed the new Jira API invocations in master
- Bumped release to 2.1.0!!!!

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -415,12 +416,12 @@ func httpAPIGetSearchIssues(ji Instance, w http.ResponseWriter, r *http.Request)
 		MaxResults: 50,
 		Fields:     []string{"key", "summary"},
 	})
+	defer CloseJiraResponse(resp)
```
Comment (jfrerich) [inline server/issue.go]: Doesn't the continue only affect the for loop?  If so, I think you can refactor and share some of code in the if/else blocks.

https://github.com/mattermost/mattermost-plugin-jira/blob/d42602692e1602d64a40b75fcc529ffe77cd9c59/server/webhook.go#L120-L130

File: server/webhook.go [hunk 2]
```diff
@@ -123,6 +123,7 @@ func (wh *webhook) PostNotifications(p *Plugin) ([]*model.Post, int, error) {
 			}
 
 			resp, _ := jiraClient.Do(req, nil)
```
Comment (jfrerich) [inline server/webhook.go]: Doesn't the continue only affect the for loop?  If so, I think you can refactor and share some of code in the if/else blocks.  You could pull out he 

https://github.com/mattermost/mattermost-plugin-jira/blob/d42602692e1602d64a40b75fcc529ffe77cd9c59/server/webhook.go#L120-L130

Comment (cpoile) [reply server/webhook.go]: That's my code, I'll make the change and commit. Good catch @jfrerich!

Comment (levb) [reply server/webhook.go]: Thanks for catching! I was search/replacing and it's really tricky with defer.

**Comment by levb** [comment]
Replaced with #218

**Pull request closed without merging**